pkg/server/endpoints/entryfetcher.go

```diff
@@ -0,0 +1,85 @@
+package endpoints
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spiffe/go-spiffe/v2/spiffeid"
+	"github.com/spiffe/spire/pkg/common/telemetry"
+	"github.com/spiffe/spire/pkg/server/api"
+	"github.com/spiffe/spire/pkg/server/cache/entrycache"
+	"github.com/spiffe/spire/pkg/server/plugin/datastore"
+	"github.com/spiffe/spire/proto/spire/types"
+	"github.com/spiffe/spire/test/clock"
+)
+
+const (
+	cacheReloadInterval = 5 * time.Second
+)
+
+var _ api.AuthorizedEntryFetcher = (*AuthorizedEntryFetcherWithFullCache)(nil)
+
+type AuthorizedEntryFetcherWithFullCache struct {
+	cache   entrycache.Cache
+	clk     clock.Clock
+	ds      datastore.DataStore
```

Comment:
I think some of the unit-tests here can be simplified if we aren't tightly coupled to entrycache.BuildFromDataStore but instead just provided this with
```
buildCache func(ctx context.Context) entrycache.Cache
```

The other positive side effect of doing that would be that this unit no longer requires the metrics and datastore, which could just be used inside of the closure the endpoints package provides for the `buildCache` function.

The benefit is that the unit-tests no longer need to populate a datastore an any meaningful way in order to hydrate the cache. In the end, what is important to this unit, is the ability to hydrate and use _some_ cache on some period basis, not specifically the full cache implementation.

Reply:
Yeah, good suggestion! I drifted a little too far into copy-paste mode on some of these unit tests, but this makes things much simpler to override in tests.